### PR TITLE
Feature/update and generate nuget

### PR DESCRIPTION
**Pull request by illanos**
Generate a nuget first day of month if vk.xml file has changed in the KhronosRegistry repository

### CODE REVIEW

File: .github/workflows/download_latest_spect_generate_nuget.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  #[push]
+  schedule:
+    - cron: "0 0 1 * *"
+
+env:
+  filePath: ${{  'vk.xml'  }}
+  fileTmpPath: ${{  'vk.xml_tmp'  }}
+  basePath: ${{  'KhronosRegistry'  }}
+  sourceUrl: ${{  'https://raw.githubusercontent.com/WaveEngine/Vulkan.NET/master/KhronosRegistry/vk.xml'  }}
+  selfUrl: ${{  'https://raw.githubusercontent.com/WaveEngine/Vulkan.NET/feature/updateAndGenerateNuget/KhronosRegistry/vk.xml'  }}
```
Comment (danielcaceresm) [inline .github/workflows/download_latest_spect_generate_nuget.yml]: This url must be from master branch

File: .github/workflows/download_latest_spect_generate_nuget.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        id: fileHash
+        run: |
+          echo "::set-output name=hash1::${{ hashFiles('vk.xml')  }}"
+          echo "::set-output name=hash2::${{ hashFiles('vk.xml_tmp')  }}" 
+          echo "::set-output name=date::$(date +'%Y.%m.%d')"
+
+  build:
+    needs: [download]
+    if: ${{ needs.download.outputs.output1 != needs.download.outputs.output2  }}
+    runs-on: windows-latest
```
Comment (danielcaceresm) [inline .github/workflows/download_latest_spect_generate_nuget.yml:45]: Check if all task can run in the same machine.

File: .github/workflows/download_latest_spect_generate_nuget.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          location: ${{ env.basePath  }}/
+      - name: Generate NuGet packages
+        run: powershell ./Generate-NuGets.ps1 -Version  ${{  needs.download.outputs.output3  }}.${{  github.run_number  }} -outputfolder ${{  env.nugetOutputPath  }}
+      - name: Publish NuGet
+        if: ${{ success() }}
+        run: |
+          cd ${{  env.nugetOutputPath  }}
+          ls *.nupkg
+          dotnet nuget push "**/*.nupkg" --skip-duplicate --no-symbols true -k ${{secrets.WAVE_NUGETORG_TOKEN}} -s https://api.nuget.org/v3/index.json
+      - name: Commit if update
```
Comment (danielcaceresm) [inline .github/workflows/download_latest_spect_generate_nuget.yml]: Change order, first commit and then publish.

File: .github/workflows/download_latest_spect_generate_nuget.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          commit_message: Updating new xr file
+          add_options: "-u"
+      - name: SendGrid Mail Action
+        if: ${{ failure() }}
+        uses: mmichailidis/sendgrid-mail-action@v1.1
+        with:
+          # The token for sendgrid
+          sendgrid-token: ${{ secrets.WAVE_SENDGRID_TOKEN }}
+          # The emails ( can be only one ) that the email will go
+          mail: mgomar@plainconcepts.com,jferrero@plainconcepts.com,jcanton@plainconcepts.com
```
Comment (danielcaceresm) [inline .github/workflows/download_latest_spect_generate_nuget.yml]: Move emails to secrets.

File: .github/workflows/download_latest_spect_generate_nuget.yml (hunk 1 above)
Comment (illanos) [reply .github/workflows/download_latest_spect_generate_nuget.yml]: done

File: .github/workflows/download_latest_spect_generate_nuget.yml (hunk 3 above)
Comment (illanos) [reply .github/workflows/download_latest_spect_generate_nuget.yml]: done

**Review by danielcaceresm — APPROVED** [review]

**Pull request merged**